test/common/router/rds_impl_test.cc

```diff
@@ -541,25 +561,25 @@ TEST_F(RouteConfigProviderManagerImplTest, Basic) {
       rds2, factory_context_, "foo_prefix");
   EXPECT_NE(provider3, provider_);
 
-  std::vector<RouteConfigProvider*> configured_providers =
-      route_config_provider_manager_->getRdsRouteConfigProviders();
-  EXPECT_EQ(2UL, configured_providers.size());
+  EXPECT_EQ(2UL,
+            route_config_provider_manager_->dumpRouteConfigs()->dynamic_route_configs().size());
```

Comment:
We only have one subscription, how can we have 2 dynamic route configs?  Should it be 1?

Reply:
we have two RDS config, `rds_` and `rds2`, they are keyed by full [Rds message](https://github.com/envoyproxy/envoy/blob/672083c7ba4e92fe9548252c45d84d583a95102d/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto#L322), not only the name, they have different cluster.